cmd/fs-v1-metadata.go

```diff
@@ -161,3 +173,39 @@ func isPartsSame(uploadedParts []objectPartInfo, completeParts []completePart) b
 	}
 	return true
 }
+
+// readFSMetadata - returns the object metadata `fs.json` content.
+func (fs fsObjects) readFSMetadata(reader io.Reader) (fsMeta fsMetaV1, err error) {
+	var metadataBytes []byte
+	metadataBytes, err = ioutil.ReadAll(reader)
+	if err != nil {
+		return fsMetaV1{}, traceError(err)
+	}
+	if len(metadataBytes) == 0 {
+		return fsMetaV1{}, traceError(io.EOF)
+	}
+
+	// Decode `fs.json` into fsMeta structure.
+	if err = json.Unmarshal(metadataBytes, &fsMeta); err != nil {
+		return fsMetaV1{}, traceError(err)
+	}
+
+	// Success.
+	return fsMeta, nil
+}
+
+// Write fsMeta to fs.json.
+func (fs fsObjects) writeFSMetadata(writer io.Writer, fsMeta fsMetaV1) error {
+	metadataBytes, err := json.Marshal(fsMeta)
+	if err != nil {
+		return traceError(err)
+	}
+	_, err = writer.Write(metadataBytes)
```

Comment:
should truncate before writing because previously uploaded object metadata might exist.